lib/src/int64.dart

```diff
@@ -199,264 +229,280 @@ class Int64 implements IntX {
+    // |....................|..................xxxx|xxxxxxxxxxxxxxxxxxxxxx| partialLowest
     // |....................|......................|......................|
-    // |....................|...................c11|......c01.............|
-    // |....................|....xxxxxxxxxxxxxxxxxx|xxxxxxxxx.............| p1
+    // |....................|.................cMPML|.....cMPL.............|
+    // |....................|....xxxxxxxxxxxxxxxxxx|xxxxxxxxx.............| partialLow
     // |....................|......................|......................|
-    // |.................c22|...............c12....|......................|
-    // |..........xxxxxxxxxx|xxxxxxxxxxxxxxxxxx....|......................| p2
+    // |................cHPL|.............cMPMH....|......................|
+    // |..........xxxxxxxxxx|xxxxxxxxxxxxxxxxxx....|......................| partialMiddle
     // |....................|......................|......................|
-    // |.................c23|..c13.................|......................|
-    // |xxxxxxxxxxxxxxxxxxxx|xxxxx.................|......................| p3
+    // |................cHPM|.cMPH.................|......................|
+    // |xxxxxxxxxxxxxxxxxxxx|xxxxx.................|......................| partialHigh
     // |....................|......................|......................|
-    // |.........c24........|......................|......................|
-    // |xxxxxxxxxxxx........|......................|......................| p4
-
-    int c00 = p0 & 0x3fffff;
-    int c01 = (p1 & 0x1ff) << 13;
-    int c0 = c00 + c01;
-
-    int c10 = p0 >> 22;
-    int c11 = p1 >> 9;
-    int c12 = (p2 & 0x3ffff) << 4;
-    int c13 = (p3 & 0x1f) << 17;
-    int c1 = c10 + c11 + c12 + c13;
-
-    int c22 = p2 >> 18;
-    int c23 = p3 >> 5;
-    int c24 = (p4 & 0xfff) << 8;
-    int c2 = c22 + c23 + c24;
+    // |........CHPH........|......................|......................|
+    // |xxxxxxxxxxxx........|......................|......................| partialHighest
+
+    int cLowPartialLow = partialLowest & 0x3fffff;
+    int cLowPartialHigh = (partialLow & 0x1ff) << 13;
+    int cLow = cLowPartialLow + cLowPartialHigh;
+
+    int cMiddlePartialLowest = partialLowest >> 22;
+    int cMiddlePartialMiddleLow = partialLow >> 9;
+    int cMiddlePartialHigh = (partialMiddle & 0x3ffff) << 4;
+    int cMiddlePartialHighest = (partialHigh & 0x1f) << 17;
+    int cMiddle = cMiddlePartialLowest +
+        cMiddlePartialMiddleLow +
+        cMiddlePartialHigh +
+        cMiddlePartialHighest;
+
+    int cHighPartialLow = partialMiddle >> 18;
+    int cHighPartialMiddle = partialHigh >> 5;
+    int cHighPartialHigh = (partialHighest & 0xfff) << 8;
+    int cHigh = cHighPartialLow + cHighPartialMiddle + cHighPartialHigh;
 
     // Propagate high bits from c0 -> c1, c1 -> c2.
-    c1 += c0 >> _BITS;
-    c2 += c1 >> _BITS;
+    cMiddle += cLow >> _lowBitCount;
+    cHigh += cMiddle >> _lowBitCount;
```

Comment:
If we are to name the bit counts by position, then we need a separate name for the two different uses of `_BITS`, e.g. `_lowBitCount` and `_middleBitCount`. Otherwise the use of `_low` here looks like a mistake.